be/src/exec/stream/state/mem_state_table.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mem_state_table.h"
+
+namespace starrocks::stream {
+
+namespace {
+
+void convert_datum_rows_to_chunk(const std::vector<DatumRow>& rows, Chunk* chunk) {
+    // Chunk should be already allocated, no need create column any more.
+    DCHECK(chunk);
+    for (size_t row_num = 0; row_num < rows.size(); row_num++) {
+        auto& row = rows[row_num];
+        for (size_t i = 0; i < row.size(); i++) {
+            VLOG_ROW << "[convert_datum_rows_to_chunk] row_num:" << row_num << ", column_size:" << row.size()
+                     << ", i:" << i;
+            DCHECK_LT(i, chunk->num_columns());
+            auto& col = chunk->get_column_by_index(i);
```

Comment:
get_column_by_index is very dangerous,  slot_id_to_idx is not stable, so can we use get_column_by_slotid

Reply:
`MemStateTable` is used only in UnitTests, and will not cross exchange.

Maybe we can use `slot_ids` later.